docs-src/app-dev-context/side-effects.md

```diff
@@ -0,0 +1,20 @@
+---
+id: side-effects
+title: Side Effects
+description: Side Effects are a method of execution to produce nondeterministic code.
+sidebar_label: Side Effects
+tags:
+  - guide-context
+---
+
+A Side Effect is a method of execution to produce nondeterministic code, such as generating a UUID or a random number.
```

Comment:
Not a method of execution. Doesn't produce code.

A decent piece from javadocs:

"Executes the provided function once, records its result into the workflow history. The recorded
   * result on history will be returned without executing the provided function during replay. This
   * guarantees the deterministic requirement for workflow as the exact same result will be returned
   * in replay."